rticonnextdds_connector_test.go

```diff
@@ -143,81 +143,171 @@ func TestDataFlow(t *testing.T) {
 	assert.Nil(t, err)
 
 	// Take any pre-existing samples from cache
-	assert.Nil(t, input.Take())
+	// err = input.Take()
 
 	xs := int8(math.MaxInt8)
 	s := int16(math.MaxInt16)
 	us := uint16(math.MaxUint16)
 	l := int32(math.MaxInt32)
-	ll := int64(123)
 	ul := uint32(math.MaxUint32)
-	ull := uint64(456)
+	// an integral value larger than 2^53 can only be retrieved or set with a dictionary/json string
+	ll := int64(math.Pow(2, 52))
+	ull := uint64(math.Pow(2, 52))
 	f := float32(math.MaxFloat32)
 	d := float64(math.MaxFloat64)
 
 	c := byte('A')
 	b := true
 	st := "test"
 
-	assert.Nil(t, output.Instance.SetInt8("xs", xs))
-	assert.Nil(t, output.Instance.SetUint8("c", c))
-	assert.Nil(t, output.Instance.SetByte("c", c))
-	assert.Nil(t, output.Instance.SetString("st", st))
-	assert.Nil(t, output.Instance.SetBoolean("b", b))
-	assert.Nil(t, output.Instance.SetInt16("s", s))
-	assert.Nil(t, output.Instance.SetUint16("us", us))
-	assert.Nil(t, output.Instance.SetInt32("l", l))
-	assert.Nil(t, output.Instance.SetUint32("ul", ul))
-	assert.Nil(t, output.Instance.SetInt("l", int(l)))
-	assert.Nil(t, output.Instance.SetInt64("ll", ll))
-	assert.Nil(t, output.Instance.SetUint64("ull", ull))
-	assert.Nil(t, output.Instance.SetUint("ul", uint(ul)))
-	assert.Nil(t, output.Instance.SetRune("l", rune(l))) //nolint //this is because "l" rune is still an int32
-	assert.Nil(t, output.Instance.SetFloat32("f", f))
-	assert.Nil(t, output.Instance.SetFloat64("d", d))
+	err = output.Instance.SetInt8("xs", xs)
+	assert.Nil(t, err)
```

Comment:
why do we split them up? 
```
	assert.Nil(t, output.Instance.SetInt8("xs", xs))
```
works the same and less lines of code too

Reply:
Updated. 
https://github.com/rticommunity/rticonnextdds-connector-go/pull/37

Reply:
Nice thanks!